waspc/src/Wasp/Generator/WebAppGenerator/RouterGenerator.hs

```diff
@@ -163,21 +159,14 @@ determineRouteTargetComponent spec (_, route) =
 createPageTemplateData :: (String, AS.Page.Page) -> PageTemplateData
 createPageTemplateData page =
   PageTemplateData
-    { _importFrom = relPathToExtSrcDir FP.</> SP.fromRelFileP (AS.ExtImport.path pageComponent),
-      _importWhat = case AS.ExtImport.name pageComponent of
-        AS.ExtImport.ExtImportModule _ -> pageName
-        AS.ExtImport.ExtImportField identifier -> "{ " ++ mkNamedImportExpr identifier pageName ++ " }"
+    { _importStmt = importStmt
     }
   where
-    relPathToExtSrcDir = "./" FP.</> SP.toFilePath extClientCodeDirInWebAppSrcDir
-
-    pageName :: String
-    pageName = fst page
-
     pageComponent :: AS.ExtImport.ExtImport
     pageComponent = AS.Page.component $ snd page
 
-mkNamedImportExpr :: String -> String -> String
-mkNamedImportExpr identifier alias
-  | identifier == alias = identifier
-  | otherwise = identifier ++ " as " ++ alias
+    importAlias :: String
+    importAlias = fst page
+
+    importStmt :: String
+    (_, importStmt) = getClientJsImportWithAlias [reldirP|./|] pageComponent $ Just importAlias
```

Comment:
I would order this in opposite way, so I can read it from top down -> importStmt, then importAlias below it. Since importStmt is more interesting.

It is an idea from CleanCode -> to write the file like a story, top down, with more high level stuff above, and low level stuff below.